src/transformers/modeling_utils.py

```diff
@@ -1373,11 +1373,10 @@ def __init__(self, config: PretrainedConfig, *inputs, **kwargs):
                 "`PretrainedConfig`. To create a model from a pretrained model use "
                 f"`model = {self.__class__.__name__}.from_pretrained(PRETRAINED_MODEL_NAME)`"
             )
-        # Save config and origin of the pretrained weights if given in model
         if not getattr(config, "_attn_implementation_autoset", False):
-            config = self._autoset_attn_implementation(
-                config, torch_dtype=torch.get_default_dtype(), check_device_map=False
-            )
+            # config usually has a `torch_dtype` but we need the next line for the `no_super_init` tests
+            dtype = config.torch_dtype if hasattr(config, "torch_dtype") else torch.get_default_dtype()
```

Comment:
this is kind of a breaking change no? 
It's kinda equivalent to #34919 as you no longer use float32, but "auto"

Reply:
oh right, we can preserve BC by force setting fp32 as dtype in configs when `torch_dtype=None` is passed. But since you are already breaking it, maybe there is no need?

Reply:
My break won't be merged until v5! let's no break and  you can merge